SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialLoadBalancingHandler.cpp

```diff
@@ -41,25 +42,48 @@ FSpatialLoadBalancingHandler::EvaluateActorResult FSpatialLoadBalancingHandler::
 
 	if (NetDriver->StaticComponentView->HasAuthority(EntityId, SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID))
 	{
-		if (!NetDriver->LoadBalanceStrategy->ShouldHaveAuthority(*Actor) && !NetDriver->LockingPolicy->IsLocked(Actor))
-		{
-			AActor* NetOwner = SpatialGDK::GetHierarchyRoot(Actor);
+		AActor* NetOwner = SpatialGDK::GetHierarchyRoot(Actor);
+		const bool bNetOwnerHasAuth = NetOwner->HasAuthority();
```

Comment:
NetOwner could be null here

Reply:
GetHierarchyRoot gives you the top of the hierarchy, so it could be the actor itself.
I added GetTopmostOwner for more clarity, which could be null.